### PR TITLE
Spain population by provinces chart

### PR DESCRIPTION
**Pull request by JTrillo**
I'm going to sumarize changes:
- I had to add two dependencies (**d3-composite-projections** & **topojson-client**) and one development dependency (**@types/topojson-client**). That's why **package.json** and **package-lock.json** are modified.
- I modified a little bit the pod **main-chart-list** to show new chart. Just that.
- I use the file **pods/spain-population-chart/spain-population-chart.config.ts** to set chart styles and other stuff like svg dimensions.
- We could create a folder to store all json files like **spain-province.json** because I think we are going to use them more than once.

### CODE REVIEW

**Comment by JTrillo** [comment]
I forgot to adjoint the source of the data: https://www.ine.es/jaxiT3/Tabla.htm?t=2852

File: pods/main-chart-list/main-chart.vm.ts [hunk 1]
```diff
@@ -11,3 +11,10 @@ export const createDefaultSampleChart = (): ChartInfoVm => ({
   picture: "static/images/default_chart.png",
   chartPageRoute: "espana-covid-marzo",
 });
+
+export const createSpainPopulationChart = (): ChartInfoVm => ({
+  title: "Spain population by provinces",
+  description: "Spain population by provinces (data updated on 1st January 2020)",
```
Comment (brauliodiez) [inline pods/main-chart-list/main-chart.vm.ts]: We have to include the source where we extracted the data

Comment (JTrillo) [reply pods/main-chart-list/main-chart.vm.ts]: True. I'm gonna fix that quickly.

Comment (JTrillo) [reply pods/main-chart-list/main-chart.vm.ts]: I'm thinking we should create a new field on the interface ***ChartInfoVm*** (**pods/main-chart-list/main-chart.vm.ts**) called *source*, puttin the source outside *description* field. Then we could use that new filed on component **chart-info-card.tsx** as a link and not like plain text (or also plain text but separeted from description).  What do you think about it?

I adjoint the current state of **chart-info-card.tsx**:
image

Comment (brauliodiez) [reply pods/main-chart-list/main-chart.vm.ts]: Source and Tags as well (tag array) we need it to get good SEO support

File: pods/spain-population-chart/spain-population-chart.config.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  strokeWidth: 1,
+  stroke: "#2f4858",
+  fill: "#008c86",
+};
+
+export const selectedProvinceStyle = {
+  strokeWidth: 2,
+  stroke: "#bc5b40",
+  fill: "#f88f70",
+};
```
Comment (brauliodiez) [inline pods/spain-population-chart/spain-population-chart.config.ts]: We can make use of classes, check the approach that is already in master if we rename the css file to .module.css we get automatically support of css modules

Comment (JTrillo) [reply pods/spain-population-chart/spain-population-chart.config.ts:23]: Something like this (?): 
image

File: pods/main-chart-list/main-chart.vm.ts (hunk 1 above)
Comment (JTrillo) [reply pods/main-chart-list/main-chart.vm.ts]: Something like this (?):
image

**Comment by JTrillo** [comment]
Ok Braulio I think you can review it again :)

**Comment by JTrillo** [comment]
Major changes:
- Created component **common-app/chart-info.component.tsx**. This component can be used inside chart component file. The new component add below the header the long description, the source and tags.
- Renamed page from english to spanish. Now file is **pages/espana-poblacion-provincias-2020.tsx**. **ChartInfoComponent** is used here.
- Short description is used on chart info cards.
- Added long description to my chart. Also created a table next to the chart on **pods/spain-population-chart/spain-population-chart.component.tsx**.

**Comment by JTrillo** [comment]
If you have any doubt please tell me.

**Comment by brauliodiez** [comment]
About the legend it should be displayed where the mouse is located (a litte bit up where the mouse point is located) and move along.

**Comment by brauliodiez** [comment]
Remove the paging on the table and add scrolling,and for instance add sorting per column, so you can decide to sort by Province or by population (if it already provides that MUI table and is easy to add it).

On the other hand population should be measured in something "Habitantes"

**Comment by brauliodiez** [comment]
About the tags, they should have a different background color

**Comment by brauliodiez** [comment]
I will open a separate case to provide support to markdown to the descriptions (no hurry with that but will make sense look ,more professional)

**Comment by brauliodiez** [comment]
Design should be responsive if you make smaller the page (e.g. mobile), spain map should take all the space and the table falldown.

File: pods/main-chart-list/main-chart.vm.ts (hunk 1 above)
Comment (brauliodiez) [reply pods/main-chart-list/main-chart.vm.ts]: That's great, I would add this as an issue and take it in a separate branch

**Comment by JTrillo** [comment]
Thank you for the feedback Braulio. I will make the changes during next week.

**Comment by JTrillo** [comment]
OK Braulio I think you can review the branch again. I modified what you said to me on comments (except provide support to markdown to the descriptions).

File: pods/spain-population-chart/components/table.component.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+function stableSort<T>(array: T[], comparator: (a: T, b: T) => number) {
+  const stabilizedThis = array.map((el, index) => [el, index] as [T, number]);
+  stabilizedThis.sort((a, b) => {
+    const order = comparator(a[0], b[0]);
+    if (order !== 0) return order;
+    return a[1] - b[1];
+  });
+  return stabilizedThis.map((el) => el[0]);
+}
```
Comment (brauliodiez) [inline pods/spain-population-chart/components/table.component.tsx:45]: Should we move this to common/sort or common/utils ?

**Review by brauliodiez — APPROVED** [review]

**Pull request merged**